mod/node-core/pkg/components/messages.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2024, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package components
+
+import (
+	"cosmossdk.io/depinject"
+	"github.com/berachain/beacon-kit/mod/async/pkg/messaging"
+	"github.com/berachain/beacon-kit/mod/async/pkg/server"
+	asynctypes "github.com/berachain/beacon-kit/mod/async/pkg/types"
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/messages"
+)
+
+type MessageServerInput struct {
+	depinject.In
+	Routes []asynctypes.MessageRoute
+}
+
+// ProvideMessageServer provides a message server.
+func ProvideMessageServer(in MessageServerInput) (
+	*server.MessageServer, error,
+) {
+	ms := server.NewMessageServer()
+	for _, route := range in.Routes {
+		if err := ms.RegisterRoute(route.MessageID(), route); err != nil {
+			return nil, err
+		}
+	}
+	return ms, nil
+}
+
+// ProvideMessageRoutes provides all the message routes.
+func ProvideMessageRoutes() []asynctypes.MessageRoute {
+	return []asynctypes.MessageRoute{
+		RouteFactory(messages.BuildBeaconBlockAndSidecars),
+		RouteFactory(messages.VerifyBeaconBlock),
+		RouteFactory(messages.FinalizeBeaconBlock),
+		RouteFactory(messages.ProcessGenesisData),
+		RouteFactory(messages.VerifySidecars),
+		RouteFactory(messages.ProcessSidecars),
+	}
+}
+
+// RouteFactory creates a new route for the given message ID.
+func RouteFactory(mID string) asynctypes.MessageRoute {
```

Comment:
i think this factory is kinda superfluous and really like dense but also maybe i just cant read